src/hypothesis/extra/numpy.py

```diff
@@ -161,19 +189,48 @@ def do_draw(self, data):
                 # values of the array across the mask.
                 one_element = np.zeros(shape=1, dtype=self.dtype)
                 one_element[0] = data.draw(self.fill)
+                fill_value = one_element[0]
+                if self.unique:
+                    try:
+                        is_nan = np.isnan(fill_value)
+                    except TypeError:
+                        is_nan = False
+
+                    if not is_nan:
+                        raise InvalidArgument(
+                            'Cannot fill unique array with non-NaN '
+                            'value %r' % (fill_value,))
+
                 np.putmask(result, needs_fill, one_element)
 
         return result.reshape(self.shape)
 
 
 @st.composite
 def arrays(
-    draw, dtype, shape, elements=None, fill=None
+    draw, dtype, shape, elements=None, fill=None, unique=False
 ):
-    """`dtype` may be any valid input to ``np.dtype`` (this includes
-    ``np.dtype`` objects), or a strategy that generates such values.  `shape`
-    may be an integer >= 0, a tuple of length >= of such integers, or a
-    strategy that generates such values.
+    """
```

Comment:
I'm going to put on my documentation hat, and ask you to do a bunch more work here:

- Start the docstring with a brief description of what the function does
- Set up and use an Intersphinx mapping so we cross-reference the Numpy docs instead of just (or not even) making them literals.  Use that here and in the other docstrings.  (And probably likewise for Pandas in the other pull)
- Build and read the docs to check formatting and consistency of punctuation (there are a few commas and periods missing)

This is not to imply that the previous docs were good (sorry!), but it's worth fixing while you're here now.